### PR TITLE
d: Fix linking shared libraries with DMD

### PR DESCRIPTION
**Pull request by ximion**
Pretty self explanatory.
We deal with DMD the same way we deal with the LDC compiler and translate the target flags so that the compiler passes them to the linker.

Fixes issues where people couldn't compile projects with multiple libraries with DMD.

### CODE REVIEW

**Comment by jpakkane** [comment]
This needs a test that checks that linking multiple libraries works.

**Comment by ximion** [comment]
@jpakkane This test would be un-runnable, as the CI is using GDC at time, which - in the version present there - doesn't support shared libraries.

**Comment by ximion** [comment]
@jpakkane Ignore my previous statement ^^ - I forgot `MESON_SKIP_TEST` exists.
I committed a testcase that allows reproducing the bug, as well as testing that it doesn't happen again.

And GDC 7.x supports shared libraries, so that issue will be solved too :-)

File: mesonbuild/backend/backends.py [hunk 1]
```diff
@@ -435,7 +435,7 @@ def build_target_link_arguments(self, compiler, deps):
         for d in deps:
             if not isinstance(d, (build.StaticLibrary, build.SharedLibrary)):
                 raise RuntimeError('Tried to link with a non-library target "%s".' % d.get_basename())
-            if isinstance(compiler, compilers.LLVMDCompiler):
+            if isinstance(compiler, compilers.LLVMDCompiler) or isinstance(compiler, compilers.DmdDCompiler):
```
Comment (QuLogic) [inline mesonbuild/backend/backends.py]: Could have been `isinstance(compiler, (compilers.LLVMDCompilter, DmdDCompiler))`.

Comment (ximion) [inline mesonbuild/backend/backends.py]: Right. Ideally someone could fix this directly in Meson with a trivial commit.

**Pull request merged**